lib/filewatcher.rb

```diff
@@ -29,7 +29,7 @@ def initialize(unexpanded_filenames, options = {})
   end
 
   def watch(&on_update)
-    %w[HUP INT TERM].each { |signal| trap(signal) { exit } }
+    %w[HUP INT TERM].each { |signal| trap(signal) { exit } if Signal.list.include? signal }
```

Comment:
What about:

```ruby
## The set of available signals depends on the OS
(%w[HUP INT TERM] & Signal.list.keys).each { |signal| trap(signal) { exit } }
```

?

Reply:
That's even better. 👍

Reply:
I think we can just don't "walk" through unsupported signals in `each`.